src/internal/connector/data_collections_test.go

```diff
@@ -98,13 +99,16 @@ func (suite *ConnectorDataCollectionIntegrationSuite) TestExchangeDataCollection
 
 	for _, test := range tests {
 		suite.T().Run(test.name, func(t *testing.T) {
+			errs := fault.New(true)
```

Comment:
this is unrelated to this PR but I did want to surface it. Maybe at some point we should make a const in the fault package that is `FailFast` so that this reads `fault.New(fault.FailFast)` instead of a random bool value. It's a small thing, but helps with readability as people don't have to go open up another package or something to figure out what the mysterious bool is